bin/warehouse/generate_warehouse.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+"""
+Creates Warehouse for all PD-Types found in archived backups.
+
+Arguments:
+fname - directory of archived backups
+operation - '-d' to compare last 2 backups (default), '-a' to compare all backups.
+"""
+import tarfile
+import sys
+import os
+import subprocess
+import shutil
+import tempfile
+from datetime import datetime
+import argparse
+
+parser = argparse.ArgumentParser(description="Run warehouse script. By default, it runs on the last 2 backups.")
+parser.add_argument("fname", help="directory of archived backups")
+parser.add_argument("-a", "--all", action='store_true', help="compare all backups.")
+args = parser.parse_args()
+
+tar_array = sorted(os.listdir(args.fname))
+if args.all == False:
+    tar_array = tar_array[-2:]
+
+prev = ''
+curr = ''
+
+def get_base(tfile):
+    base = os.path.basename(tfile)
+    pd_name = os.path.splitext(os.path.splitext(base)[0])[0]
+    return pd_name
+
+def extract(tfile, dest):
+    fpath = './' + dest
+    tar = tarfile.open(args.fname + tfile)
+    tar.extractall(path=fpath)
+    tar.close()
+    return fpath
+
+def run_migrations(fpath, temp_dir):
+
+    for csvfile in os.listdir(fpath):
+        print("Migrating {0} from directory {1}".format(csvfile, fpath))
+        proc = subprocess.Popen(['python', 'migrate_all.py', fpath+'/'+csvfile, temp_dir+'/'+fpath+'m_'+csvfile])
+        if proc.wait() == 1:
+            sys.exit()

```

Comment:
```suggestion
        if proc.wait():
            sys.exit(1)

```
is better because the exit code could be any non-zero value and we want to return a failure code from our own script.
same comment for below